Pull Request for feature/previous work

exoticDFT opened the pull request:
- Adds Python codebase to the toolbox.
- Adds several Python modules for defining useful Carla API wrapper functions.
  - Adds actor module for handling functions that deal with Carla Actor objects.
  - Adds camera module, within a sensors module, for handling Carla Camera Sensors.
  - Adds client module for handling functions that deal with a Carla Client object.
  - Adds common module for handling functions that deals with general functions, not necessarily Carla related.
  - Adds world module for handling functions that manipulate the Carla World.
- Adds some unit tests for Python code.
- Updates README for new Python codebase.

exoticDFT committed:
Begins the commits of collection of previous Carla wrappers

- Adds folder structure for Python base Carla wrapper code.
- Creates the utils.client module for the toolbox.

exoticDFT committed:
Adds testing for the python modules

- Adds some testing, specifically for the utils.client module.

exoticDFT committed:
Adds actor module to the utils module

- Adds code to handle actor related wrappers to the utils module.

exoticDFT committed:
Adds world module to the utils module

- Adds code to handle world related wrappers to the utils module.

exoticDFT committed:
Adds first example for using toolbox's Python wrappers

- Adds an example to spawn a vehicle using the Python wrappers in the toolbox.

exoticDFT committed:
Updates the utils.client module

- Adds a reset parameter to the create function to reset the world on client initialization.

exoticDFT committed:
Updates the utils.world module

- Adds a new function to destroy all "dynamic" actors in the Carla world.

exoticDFT committed:
Adds a new common module in the utils module.

- Adds a new file for common wrapper functions.

exoticDFT committed:
Updates the client module

- Adds a function to remove all actors in a provided list through the Client's apply_batch method.

exoticDFT committed:
Updates the examples

- Updates the spawn vehicle example to use the common module, improve the execution and use the new client module destroy actors function.

exoticDFT committed:
Updates the client module

- Adds new function to start the client recorder.
- Adds new function to stop the client from recording.

exoticDFT committed:
Cleans up the client module

- Cleans up the client module by adding typing to the function parameters and output.
- Updates the docstrings for the functions.

exoticDFT committed:
Cleans up the world module

- Cleans up the world module by adding typing to the function parameters and output.
- Updates the docstrings for the functions.

exoticDFT committed:
Cleans up the actor module

- Cleans up the actor module by adding typing to the function parameters and output.
- Updates the docstrings for the functions.

exoticDFT committed:
Updates world module

- Adds parameter to spawn_actor function to toggle autopilot.
- Adds functionality to spawn_actor function to turn on autopilot for vehicles.
- Updates example to use the new autopilot functionality.

exoticDFT committed:
Cleans up the common module

- Cleans up the common module by adding typing to the function parameters and output.
- Updates the docstrings for the functions.
- Renames one function to match the parameters.

exoticDFT committed:
Adds functions to the common module

- Adds a new function to translate and rotate a transform by the provided offsets.
- Adds a new function to translate a transform by the provided offset.
- Updates the docstrings for the functions.

exoticDFT committed:
Adds new function to world module

- Adds a function to change the weather in the world module.
- Updates formatting.

exoticDFT committed:
Creates a new module for cameras under a new sensors module

- Creates a new module for sensor related code.
- Creates a new module for camera related code under the sensors module.

exoticDFT committed:
Updates the gitignore file

- Adds Python cache files to the ignore list.

exoticDFT committed:
Adds new wrapper functions to world module

- Adds a function to draw a line between two actors in the Carla server world.
- Adds a function to draw an arc between two actors in the Carla server world.

pep8speaks commented:
Hello @exoticDFT! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:















There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2020-05-30 05:01:30 UTC

exoticDFT committed:
Updates README

- Updates the README to add how to call the Python example.
- Adds extra time to timeout of the example, just in case the server/client are being slow.

exoticDFT committed:
Adds some documentation for the Python examples

- Adds a README to the Python examples folder to begin documenting the examples.
- Adds new documentation with some details of the first Python example.

exoticDFT committed:
Finalizes the documentation for the Python example

- Finishes the documentation for the spawn_vehicle.py example.
- Updates the example to make it a little cleaner.

exoticDFT committed:
Updates formatting to match PEP8 standard

- Updates several minor formatting issues that did not meet the PEP8 standard.
